org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -3337,6 +3355,22 @@ public static List<String> getAllVersions() {
 		return allVersions;
 	}
 
+	/**
+	 * Returns all Java source versions fully supported by Eclipse compiler in the order of their introduction. For
+	 * example, {@link JavaCore#VERSION_1_8} appears before {@link JavaCore#VERSION_10}.
+	 * <p>
+	 * Note, some not included older or newer Java versions might be known by Eclipse compiler internally but not
+	 * exposed via this API because compiler does not support these anymore (or yet).
+	 *
+	 * @return all Java source versions fully supported by Eclipse compiler
+	 * @see #isJavaSourceVersionSupportedByCompiler(String)
+	 * @see #getFirstJavaSourceVersionSupportedByCompiler()
+	 * @since 3.39
+	 */
+	public static List<String> getAllJavaSourceVersionsSupportedByCompiler() {
```

Comment:
Have you considered to use a `SequencedSet<String>` here?
As far as I can see the uses of this method either fetch the first or last element or test if a String is contained in the list. The latter would be faster if a `LinkedHashSet` were used, which implements `SequencedSet`. That interface also provide `getFirst()` and `getLast()` and therefore seems to be an ideal fit for this use-case.

The only 'problem' is that `SequencedSet` was only added in Java-21. But since SimRel permits Java-21 now this might be a good opportunity to move JDT (at least parts of it) to Java-21 now instead of waiting for a few months more and having a List-based API forever?

Reply:
Yes, I considered, but I didn't wanted to change JRE requirement because of that.

Reply:
OK, thanks.
That's relatable, but IMHO one more reason pro moving the SDK to Java-21 in the SDK.

Reply:
By chance I was remembered that we already have `SortedSet` (which inherits from `SequencedSet` in Java-21) and I think that would be a good fit as well:
Please have a look at https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2770